unit-10/javascrip/main.js

```diff
@@ -0,0 +1,104 @@
+// lesson 1
+function printMultiplicationTable(number) {
+  if (typeof(number) !== 'number') {
+    return 'Data nhập vào phải là số'
+  }
+  for (let i = 1; i <= 10; i++) {
+    console.log(number + ' x ' + i + ' = '+ number * i )
+  }
+}
+
+// lesson 2
+function getEvenNumber(n) {
+  if (typeof(n) !== 'number' || n < 0 || n > 30) {
+    return 'Data nhập vào phải là số nguyên dương lớn hơn 0 và nhỏ hơn 30'
+  }
+  for (let i = 2; i <= n; i += 2) {
+    console.log(i);
+  }
+}
+
+// lesson 3
+function getNumberSum(n) {
+  let sum = 0
+  if (typeof(n) !== 'number' || n < 0 || n > 30) {
+    return 'Data nhập vào phải là số nguyên dương lớn hơn 0 và nhỏ hơn 30'
+  }
+  for (let i = 1; i <= n; i++) {
+    sum += i
+  }
+  console.log('Tổng = ' + sum)
+}
```

Comment:
```suggestion
// lesson 3
const checkIsNotNumber = number => typeof(number) !== 'number' || number < 0 || number > 30
function getNumberSum(number) {
  const isNotNumber = checkIsNotNumber(number)
  if (isNotNumber) {
    return '....'
  }

  let sum = 0
  for (let i = 1; i <= number; i++) {
    sum += i
  }
  return sum
}
```

1 logic dùng đi dùng lại. nhiều chỗ thì viết tách ra thành. 1. hàm